chromium_src/tools/grit/preprocess_if_expr.py

```diff
@@ -44,17 +44,42 @@ def maybe_keep_upstream_version(override_in_folder, out_folder, override_file):
     return None
 
 
+def run_mangler(mangler_file, preprocess_file):
+    """Runs the mangler on the given file"""
+    import subprocess
+
+    ts_config = os.path.join(get_src_dir(), 'brave', 'tsconfig.json')
+    lit_mangler = os.path.join(get_src_dir(), 'brave', 'build', 'commands',
+                               'scripts', 'lit_mangler.ts')
+
+    # Note: We read from and write to the preprocess file - this way any
+    # preprocessing that upstream does will be mangled.
+    subprocess.run([
+        'npx', '--no', '--', 'tsx', '--tsconfig', ts_config, lit_mangler,
+        'mangle', '-m', mangler_file, '-i', preprocess_file, '-o',
+        preprocess_file
+    ],
+                   check=True)
+
```

Comment:
Seems like injecting the mangle script run after preprocess is the simplest and most universal place. And if we want to type check we could run `tsc` here. Even though it's going to run separately for each file, it should be fast?
But it seems like we _could_ do this at a higher level as part of the `build_webui` GN template after the preprocess stage, since all lit .html.ts files will run through that target. Not sure if doing it in GN will help with any issues that come up in review.
Also, if we're having any performance issues perhaps we could run against all the files in the directory or whatever list `preprocess_if_expr` has. But you'd need to turn the module into a class or similar.

Reply:
Yeah, I'm not sure what the best solution here is - at least with this, it will only be rerun when the target the manglers exist in change, whereas if this was part of `build_webui` it would have to run whenever any of the manglers changed, which I'm not sure would be better.

Additionally, it would make ordering quite difficult as this needs to run after the Chromium files are preprocessed (so it can run over the preprocessed files and doesn't get overwritten) but before the Typescript build happens.